controllers/front/successHosted.php

```diff
@@ -76,7 +76,8 @@ public function postProcess()
             if (
                 !$authResponseBody->getLiability()->getLiabilityShift() &&
                 in_array($order->payment, SaferPayConfig::SUPPORTED_3DS_PAYMENT_METHODS) &&
-                $paymentBehaviourWithout3DS === SaferPayConfig::PAYMENT_BEHAVIOR_WITHOUT_3D_CANCEL
+                $paymentBehaviourWithout3DS === SaferPayConfig::PAYMENT_BEHAVIOR_WITHOUT_3D_CANCEL ||
+                $authResponseBody->getTransaction()->getStatus() === 'CANCELED'
```

Comment:
Is it the only order state that should cancel order? Maybe better solution would be if current state is not in successful order states array?
Also, don't hardcode states, I'm sure that there are constants for this.